src/peer_manager.rs

```diff
@@ -486,13 +528,17 @@ impl PeerManager {
     }
 
     /// Marks the given peer as "connected and waiting for `NodeIdentify`".
-    pub fn connected_to(&mut self, peer_id: &PeerId) -> bool {
-        self.set_state(peer_id, PeerState::AwaitingNodeIdentify(false))
+    pub fn connected_to(&mut self, peer_id: &PeerId) {
+        if !self.set_state(peer_id,
+                           PeerState::AwaitingNodeIdentify((false, Instant::now()))) {
+            let _ = self.unknown_peers.insert(peer_id.clone(), Instant::now());
+        }
     }
 
     /// Marks the given peer as "connected via tunnel and waiting for `NodeIdentify`".
     pub fn tunnelling_to(&mut self, peer_id: &PeerId) -> bool {
-        self.set_state(peer_id, PeerState::AwaitingNodeIdentify(true))
+        self.set_state(peer_id,
```

Comment:
Should we also provide the same mechanism here? Currently `states/node.rs` relies on the returned boolean value.